posts/2023-08-22-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,156 @@
+```
+This will create a new namespace named argocd as well as apply the install manifests to that namespace.  It alternatively can be installed via OLM.
+
+
+== Using ArgoCD ==
+Get the default Admin password using the following command:
+```
+kubectl get secret argocd-initial-admin-secret -n argocd -o jsonpath='{.data.password}' | base64 -d
+```
+Next, go to the route of the `argocd-server` and log with the `admin` user and password.
+[.img_border_light]
+image::/img/blog/configuration-drift-1.png[Argo Login,width=70%,align="center"]
+
+Once logged, in you should see this page:
+[.img_border_light]
+image::/img/blog/configuration-drift-2.png[Argo Home,width=70%,align="center"]
+
+== Creating a repository to be used for GitOps ==
+Now that you've successfully installed ArgoCD and gained access to its console, the next step is to set up a GitHub repository for Argo CD to synchronize with.
+
+For the purposes of this blog, we'll use the Daytrader7 sample application. This application is built around the concept of an online stock trading system, and the Daytrader7 README provides a comprehensive overview.
+
+It's a recommended best practice to maintain separation between configuration repositories and code repositories. Keeping these separate ensures that infrastructure changes are decoupled from application changes, allowing for more granular control, better audit trails, and reduced risk of unintended side-effects. For this example, we'll create a deploy directory within our existing repository to house the deployment configuration.
+
+To begin, navigate to GitHub and fork this repository: https://github.com/WASdev/sample.daytrader7. Use your forked repository when configuring ArgoCD in the following steps
+
+We assume that your CI pipleline has a way to build contianer image for your application. For this example, we have a pre-built cotnainer image used in the deployment files `deploy/daytrader7-deploy.yaml`.
+
+== Deploying DayTrader 7 via GitOps ==
+
+Now that we have the Git configuration repository ready for deployment, it’s time to configure ArgoCD to deploy the application.
+
+Go to the Argo CD console and ensure that you are logged in.
+Click the New App button near the top left of the console.
+For Application name, let’s call this one daytrader7, keeping it in the default Argo project and setting the sync policy to Manual. Check the Auto-Create Namespace box as well.
+[.img_border_light]
+image::/img/blog/configuration-drift-3.png[Creating the App,width=70%,align="center"]
+Scroll down to the Source and change the Repository URL to your newly forked repo. Change the path, which is the path where Argo CD is looking for deployment files, to deploy. Set the Cluster URL to https://kubernetes.default.svc, which is the local cluster URL. Then set the namespace to daytrader7
+Click Create in the top left.
+[.img_border_light]
+image::/img/blog/configuration-drift-4.png[Setting the Repo,width=70%,align="center"]
+
+== Syncing DayTrader ==
```

Comment:
This subtitle doesn't give me much of an idea as to how this relates to the overall task of using GitOps to deploy my app. Can we make this subtitle clearer? why is syncing needed? this reason could be the name of the section and would help me to understand where we are on this journey and why I should read this section.